cep18/src/main.rs

```diff
@@ -360,29 +369,263 @@ pub extern "C" fn change_security() {
         sec_change_map: badge_map,
     }));
 }
+
+/// Entrypoint to migrate user and contract keys regarding balances provided as argument from 1.x to
+/// 2.x key/hash storage version. Argument is a single BTreeMap<Key, bool>. The key is the already
+/// stored key in the system (previously Key::Hash or Key::Account), while the bool value is the
+/// verification for the key's flag (true for account, false for contract).
+/// Going forward these will be stored are Key::AddressableEntity(EntityAddr::Account)
+/// and Key::AddressableEntity(EntityAddr::SmartContract) respectively.
+#[no_mangle]
+pub fn migrate_users_balance_keys() {
+    let event_on: bool = get_named_arg(EVENTS);
+    let revert_on: bool = get_named_arg(REVERT);
+    let mut success_map: Vec<(Key, Key)> = Vec::new();
+    let mut failure_map: BTreeMap<Key, String> = BTreeMap::new();
+
+    let keys: BTreeMap<Key, bool> = get_named_arg(USER_KEY_MAP);
+    let balances_uref = get_balances_uref();
+    for (old_key, flag) in keys {
```

Comment:
Would you rename generic "flag" term to "is_account" maybe ? or key_type_is_account ? is_account_key ?